### PR TITLE
Update EditCommand in Developer Guide

### PR DESCRIPTION
**Pull request by underthehai**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -225,6 +225,59 @@ Aspect: How navigation between directory works
...
+
+EditCommand must have at least one field that is edited. Any fields that are invalid will throw an error. 
+Hence, child classes of EditCommand accounts for editing valid types of field. Each of the following classes extends `EditCommand`
+- `EditActivityCommand`
+- `EditAccommodationCommand`
+- `EditFriendCommand`
+- `EditTravelPlanCommand`
+
+`EditCommand` accounts for duplicated objects.
+Two objects are considered the same when they have the same name.
```
Comment (jiaweiteo) [inline docs/DeveloperGuide.md:254]: I think the accomodation class requires start date and end date to be the same too for two accommodation object to be the same.

**Review by jiaweiteo — APPROVED** [review]
LGTM, just a minor change to the "same" definition.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (underthehai) [reply docs/DeveloperGuide.md]: yup, travelplan too. I updated them

**Review by lyeyixian — APPROVED** [review]
LGTM

**Pull request merged**